### PR TITLE
Fix web notification icon not updated once you read all notifications

### LINKED ISSUE #29065: Web notification icon not updated once you read said notifications
### Description

Gitea is really good about updating browser tabs when new notifications arrive. When a notification comes in, ALL of my browser tabs, open to the gitea instance in question, properly update the notification bell. However, once you read the notifications the other tabs **do not** update even after waiting hours.

This is extremely confusing as one typically has several gitea tabs open at once and some will say 0 notifications while the others are in some mixed state of having a badge that says 1, 2, or 5 unread notifications etc.

### Gitea Version

1.21.2+426-g13e26f04b

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

Tab 1
notif-0unread

Tab 2..n
notif-1unread_allothertabs

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

Own build - https://projects.blender.org/

### Database

SQLite

### PR DESCRIPTION
**Pull request by kiatt210**
Fix #29065
Remove status filtering from GetUIDsAndNotificationCounts sql.

### CODE REVIEW

**Comment by wxiaoguang** [comment]
Could you add some details about how it fixes the problem? TBH I haven't caught the point 🤣

**Comment by kiatt210** [comment]
I'm sorry, of course 🙂

When the number of unread notifications becomes 0, the original SQL query does not return the user's row. 
As a result, there will not be a new notification-count event to update the icon. With this modification, the user's row will be included in the result with 0 count, and the event will be triggered.

[The event generation ](https://github.com/go-gitea/gitea/blob/1a811c0bd10723ada73eb1c48a28cbc0775c6749/modules/eventsource/manager_run.go#L74-L83):
```golang
uidCounts, err := activities_model.GetUIDsAndNotificationCounts(ctx, then, now)
if err != nil {
	log.Error("Unable to get UIDcounts: %v", err)
}
for _, uidCount := range uidCounts {
	m.SendMessage(uidCount.UserID, &Event{
		Name: "notification-count",
		Data: uidCount,
	})
}
```

**Review by wxiaoguang — APPROVED** [review]
Overall LGTM

The details could be written into comments.

**Review by wxiaoguang — COMMENTED** [review]
Hmm ...  I see another solution, to keep the old SQL (to avoid performance regression since there are some indices).

A new approach could be like this:

* Use the old SQL
* Append non-existing UIDs into the result

It seems safer than changing the SQL

**Comment by kiatt210** [comment]
I need to find out how to get the missing UIDs, but I am going to try. 

Thanks for the idea!

**Comment by wxiaoguang** [comment]
Read the code again, it seems that it is impossible to figure out the non-existing IDs quickly.

So current method should be good enough. Let's try it.

**Review by wxiaoguang — APPROVED** [review]

**Review by lunny — APPROVED** [review]

File: models/activities/notification.go [hunk 1]
```diff
@@ -288,11 +288,11 @@ type UserIDCount struct {
 
 // GetUIDsAndNotificationCounts between the two provided times
```
Comment (wxiaoguang) [inline models/activities/notification.go]:
```suggestion
// GetUIDsAndNotificationCounts returns the unread counts for every user between the two provided times.
// It must return all user IDs which appear during the period, including count=0 for users who have read all.
```

**Pull request merged**